.github/workflows/main.yml

```diff
@@ -40,6 +40,31 @@ env:
 
 
 jobs:
+  check_changed_folders:
+    name: Check files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get changed files in the _distutils folder
+        id: changed-files-specific-distutils
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            setuptools/_distutils/**
+      - name: Get changed files in the _vendor folder
+        id: changed-files-specific-vendor
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            setuptools/_vendor/**
+      - name: Run step if any file(s) in the docs folder change
+        if: steps.changed-files-specific-distutils.outputs.any_changed == 'true' || steps.changed-files-specific-vendor.outputs.any_changed == 'true'
+        run: |
+          echo "One or more files in the setuptools/_distutils or setuptools/_vendor folder has changed."
+          exit 1
```

Comment:
What should happen when a maintainer is updating one of these vendored dependencies? Do they manually ignore the failure?

Reply:
Yeah, that would be an expected failure.